staging/src/k8s.io/component-helpers/resource/helpers.go

```diff
@@ -44,15 +45,66 @@ type PodResourcesOptions struct {
 	// NonMissingContainerRequests if provided will replace any missing container level requests for the specified resources
 	// with the given values.  If the requests for those resources are explicitly set, even if zero, they will not be modified.
 	NonMissingContainerRequests v1.ResourceList
+	// PodLevelResourcesEnabled indicates that the pod-level resources feature gate
+	// is enabled.
+	PodLevelResourcesEnabled bool
 }
 
-// PodRequests computes the pod requests per the PodResourcesOptions supplied. If PodResourcesOptions is nil, then
-// the requests are returned including pod overhead. The computation is part of the API and must be reviewed
-// as an API change.
+var supportedPodLevelResources = sets.New(v1.ResourceCPU, v1.ResourceMemory)
+
+func GetSupportedPodLevelResources() sets.Set[v1.ResourceName] {
+	return supportedPodLevelResources
+}
+
+// IsSupportedPodLevelResources checks if a given resource is supported by pod-level
+// resource management through the PodLevelResources feature. Returns true if
+// the resource is supported.
+func IsSupportedPodLevelResource(name v1.ResourceName) bool {
+	return supportedPodLevelResources.Has(name)
+}
+
+// IsPodLevelResourcesSet check if PodLevelResources feature is enabled and
+// pod-level resources are set.
+func IsPodLevelResourcesSet(pod *v1.Pod, podLevelResourcesEnabled bool) bool {
+	return podLevelResourcesEnabled && pod.Spec.Resources != nil
```

Comment:
changed it to:
return pod.Spec.Resources != nil &&
		(len(pod.Spec.Resources.Requests) > 0 ||
			len(pod.Spec.Resources.Limits) > 0)

It is confusing? Added a comment for clarity. Also wanted to remove dependency on feature gate since it will always be true in Beta and GA.